File: App/DesignSystem/ISG/NovaWorkPermitLibraryScreen.swift
import SwiftUI
import UniformTypeIdentifiers

private struct NovaWorkPermitWordDocument: FileDocument {
    static let docxType = UTType(filenameExtension: "docx") ?? .data
    static var readableContentTypes: [UTType] { [docxType] }
    let data: Data

    init(data: Data) { self.data = data }
    init(configuration: ReadConfiguration) throws {
        data = configuration.file.regularFileContents ?? Data()
    }
    func fileWrapper(configuration: WriteConfiguration) throws -> FileWrapper {
        FileWrapper(regularFileWithContents: data)
    }
}

private struct NovaWorkPermitTemplate: Decodable, Identifiable {
    let code: String
    let title: String
    let usage: String
    let jobs: [String]
    let sectors: [String]
    let filename: String
    let note: String

    var id: String { code }

    var searchText: String {
        ([code, title, usage, note] + jobs + sectors).joined(separator: " ")
    }

    var fileURL: URL? {
        let base = String(filename.dropLast(".docx".count))
        return Bundle.main.url(forResource: base, withExtension: "docx",
                               subdirectory: "WorkPermitAssets/work_permits")
            ?? Bundle.main.url(forResource: base, withExtension: "docx")
    }
}

/// A read-only, bundled sample library. No company record or permit approval is created here.
struct NovaWorkPermitLibraryScreen: View {
    let onBack: () -> Void
    @State private var query = ""
    @State private var sector: String?
    @State private var job: String?
    @State private var openFilter: String?
    @State private var exportDocument: NovaWorkPermitWordDocument?
    @State private var exportName = ""
    @State private var exporting = false
    @State private var exportError: String?

    private let templates: [NovaWorkPermitTemplate] = {
        guard let url = Bundle.main.url(forResource: "catalog", withExtension: "json",
                                        subdirectory: "WorkPermitAssets/work_permits")
                ?? Bundle.main.url(forResource: "catalog", withExtension: "json"),
              let data = try? Data(contentsOf: url),
              let entries = try? JSONDecoder().decode([NovaWorkPermitTemplate].self, from: data)
        else { return [] }
        return entries.sorted { $0.code < $1.code }
    }()

    private var sectors: [String] { Array(Set(templates.flatMap(\.sectors)).subtracting(["Tüm sektörler"])).sorted() }
    private var jobs: [String] { Array(Set(templates.flatMap(\.jobs))).sorted() }

    private func searchKey(_ value: String) -> String {
        value.folding(options: [.caseInsensitive, .diacriticInsensitive], locale: Locale(identifier: "tr_TR"))
            .replacingOccurrences(of: "ı", with: "i")
    }

    private var visible: [NovaWorkPermitTemplate] {
        let words = searchKey(query).split(whereSeparator: \.isWhitespace).map(String.init)
        return templates.filter { item in
            (sector == nil || item.sectors.contains(sector!) || item.sectors.contains("Tüm sektörler"))
                && (job == nil || item.jobs.contains(job!))
                && words.allSatisfy { searchKey(item.searchText).contains($0) }
        }
    }

    var body: some View {
        NovaPageSurface(onEdgeBack: onBack) {
            ScrollView {
                LazyVStack(alignment: .leading, spacing: 12) {
                    NovaPageHeading(title: RDLocalization.string("localizable.nova.work.permit.library.screen.calisma.izni.ornekleri.c37a490d", table: .localizable, fallback: "Çalışma İzni Örnekleri"), onBack: onBack)
                    NovaHelpHint(text: RDLocalization.string("localizable.nova.work.permit.library.screen.56.duzenlenebilir.word.ornegi.uygun.formu.arayin.a258ccc9", table: .localizable, fallback: "56 düzenlenebilir Word örneği. Uygun formu arayın, indirin ve kendi saha prosedürünüze göre uyarlayın. Bu örnekler çalışma onayı veya izin kaydı oluşturmaz."))
                    HStack(spacing: 8) {
                        Image(systemName: "magnifyingglass")
                        TextField(RDLocalization.string("localizable.nova.work.permit.library.screen.form.is.veya.kelime.ara.4a1cdb74", table: .localizable, fallback: "Form, iş veya kelime ara"), text: $query)
                            .accessibilityIdentifier("permit.search")
                    }
                    .padding(12)
                    .background(.white, in: RoundedRectangle(cornerRadius: 14))
                    HStack(spacing: 8) {
                        NovaFileChooserButton(label: RDLocalization.string("localizable.nova.work.permit.library.screen.sektor.40d89aaf", table: .localizable, fallback: "Sektör"), value: sector ?? "Tüm sektörler",
                            isOpen: openFilter == "sector", identifier: "permit.sector") {
                            openFilter = openFilter == "sector" ? nil : "sector"
                        }
                        NovaFileChooserButton(label: RDLocalization.string("localizable.nova.work.permit.library.screen.is.grubu.c3856e06", table: .localizable, fallback: "İş grubu"), value: job ?? "Tüm işler",
                            isOpen: openFilter == "job", identifier: "permit.job") {
                            openFilter = openFilter == "job" ? nil : "job"
                        }
                    }
                    if let openFilter {
                        NovaFileChooserPanel(
                            options: openFilter == "sector"
                                ? [.init(id: nil, title: RDLocalization.string("localizable.nova.work.permit.library.screen.tum.sektorler.acccefbc", table: .localizable, fallback: "Tüm sektörler"))] + sectors.map { .init(id: $0, title: $0) }
                                : [.init(id: nil, title: RDLocalization.string("localizable.nova.work.permit.library.screen.tum.isler.495911c9", table: .localizable, fallback: "Tüm işler"))] + jobs.map { .init(id: $0, title: $0) },
                            selected: openFilter == "sector" ? sector : job,
                            identifier: "permit.\(openFilter).options") { value in
                            if openFilter == "sector" { sector = value } else { job = value }
                            self.openFilter = nil
                        }
                    }
                    NovaText(text: RDLocalization.format("localizable.nova.work.permit.library.screen.1.2.ornek.form.27f33b22", table: .localizable, fallback: "%1$@ / %2$@ örnek form", arguments: [String(describing: visible.count), String(describing: templates.count)]), style: .metaQuiet)
                    if let exportError { NovaText(text: exportError, style: .metaQuiet) }
                    if templates.isEmpty {
                        NovaEmptyState(title: RDLocalization.string("localizable.nova.work.permit.library.screen.formlar.yuklenemedi.2d408e6c", table: .localizable, fallback: "Formlar yüklenemedi"), message: RDLocalization.string("localizable.nova.work.permit.library.screen.uygulama.paketindeki.word.katalogu.bulunamadi.78218506", table: .localizable, fallback: "Uygulama paketindeki Word kataloğu bulunamadı."))
                    } else if visible.isEmpty {
                        NovaEmptyState(title: RDLocalization.string("localizable.nova.work.permit.library.screen.uygun.form.bulunamadi.27545fa2", table: .localizable, fallback: "Uygun form bulunamadı"), message: RDLocalization.string("localizable.nova.work.permit.library.screen.arama.kelimesini.veya.filtreleri.degistirin.beb9d84b", table: .localizable, fallback: "Arama kelimesini veya filtreleri değiştirin."))
                    } else {
                        ForEach(visible) { item in
                            NovaCard(padding: 12) {
                                HStack(alignment: .center, spacing: 10) {
                                    VStack(alignment: .leading, spacing: 4) {
                                        NovaText(text: "\(item.code) · \(item.title)", style: .bodyStrong)
                                            .lineLimit(2)
                                        NovaText(text: item.usage, style: .meta)
                                            .lineLimit(1)
                                        NovaText(text: item.jobs.joined(separator: " · "), style: .metaQuiet)
                                            .lineLimit(1)
                                    }
                                    .frame(maxWidth: .infinity, alignment: .leading)
                                    if let url = item.fileURL {
                                        Button { beginExport(item, url: url) } label: {
                                            Image(systemName: "arrow.down.to.line")
                                                .font(.system(size: 18, weight: .semibold))
                                                .frame(width: 44, height: 44)
                                                .background(Color.black.opacity(0.045), in: RoundedRectangle(cornerRadius: 12))
                                        }
                                        .buttonStyle(.plain)
                                        .accessibilityLabel(RDLocalization.format("localizable.nova.work.permit.library.screen.1.word.dosyasini.indir.17a3d8f1", table: .localizable, fallback: "%1$@ Word dosyasını indir", arguments: [String(describing: item.title)]))
                                        .accessibilityIdentifier("permit.download.\(item.code)")
                                    } else {
                                        Image(systemName: "exclamationmark.triangle")
                                            .accessibilityLabel(RDLocalization.string("localizable.nova.work.permit.library.screen.word.dosyasi.bulunamadi.6854465e", table: .localizable, fallback: "Word dosyası bulunamadı"))
                                    }
                                }
                            }
                        }
                    }
                }
                .padding(.horizontal, 16)
                .padding(.top, 4)
                .padding(.bottom, novaTabBarInset)
            }
        }
        .fileExporter(isPresented: $exporting, document: exportDocument,
                      contentType: NovaWorkPermitWordDocument.docxType,
                      defaultFilename: exportName) { result in
            if case .failure(let error) = result { exportError = error.localizedDescription }
            exportDocument = nil
        }
    }

    private func beginExport(_ item: NovaWorkPermitTemplate, url: URL) {
        do {
            exportDocument = NovaWorkPermitWordDocument(data: try Data(contentsOf: url))
            exportName = String(item.filename.dropLast(".docx".count))
            exportError = nil
            exporting = true
        } catch {
            exportError = "Word dosyası açılamadı. Tekrar deneyin."
        }
    }
}

File: android/feature/nova/src/main/kotlin/com/riskdetectedan/feature/nova/NovaWorkPermitLibraryScreen.kt
package com.riskdetectedan.feature.nova

import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.IconButton
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.riskdetectedan.core.designsystem.isg.*
import kotlinx.serialization.json.*
import java.text.Normalizer
import java.util.Locale

private data class WorkPermitSample(
    val code: String, val title: String, val usage: String, val sectors: List<String>,
    val jobs: List<String>, val filename: String, val note: String,
) {
    val searchText = (listOf(code, title, usage, note) + sectors + jobs).joinToString(" ")
}

private fun workPermitCatalog(raw: String): List<WorkPermitSample> = Json.parseToJsonElement(raw).jsonArray.map { element ->
    val item = element.jsonObject
    fun value(key: String) = item.getValue(key).jsonPrimitive.content
    fun values(key: String) = item.getValue(key).jsonArray.map { it.jsonPrimitive.content }
    WorkPermitSample(value("code"), value("title"), value("usage"), values("sectors"), values("jobs"), value("filename"), value("note"))
}.sortedBy { it.code }

private fun permitSearchKey(value: String): String = Normalizer.normalize(value.lowercase(Locale.forLanguageTag("tr-TR")), Normalizer.Form.NFD)
    .replace(Regex("\\p{Mn}+"), "").replace('ı', 'i')

/** Bundled Word examples only; saving a file never creates or approves a permit record. */
@Composable
fun NovaWorkPermitLibraryScreen(onBack: () -> Unit) {
    val context = LocalContext.current
    LaunchedEffect(Unit) { com.riskdetectedan.core.data.nova.NovaForYouService.recordUse("work_permit_forms") }
    val catalog = remember(context) { runCatching {
        context.assets.open("work_permits/catalog.json").bufferedReader(Charsets.UTF_8).use { workPermitCatalog(it.readText()) }
    }.getOrDefault(emptyList()) }
    var query by remember { mutableStateOf("") }
    var sector by remember { mutableStateOf<String?>(null) }
    var job by remember { mutableStateOf<String?>(null) }
    var openFilter by remember { mutableStateOf<String?>(null) }
    var pending by remember { mutableStateOf<WorkPermitSample?>(null) }
    var message by remember { mutableStateOf<String?>(null) }
    val save = rememberLauncherForActivityResult(ActivityResultContracts.CreateDocument("application/vnd.openxmlformats-officedocument.wordprocessingml.document")) { uri ->
        val entry = pending
        pending = null
        if (uri != null && entry != null) {
            message = runCatching {
                require(entry.filename.startsWith(entry.code + "_") && '/' !in entry.filename)
                context.assets.open("work_permits/${entry.filename}").use { source ->
                    context.contentResolver.openOutputStream(uri)?.use { target -> source.copyTo(target) }
                        ?: error("Unable to open destination")
                }
                "Word dosyası kaydedildi."
            }.getOrElse { "Word dosyası kaydedilemedi." }
        }
    }
    val sectors = remember(catalog) { catalog.flatMap { it.sectors }.distinct().filterNot { it == "Tüm sektörler" }.sorted() }
    val jobs = remember(catalog) { catalog.flatMap { it.jobs }.distinct().sorted() }
    val terms = remember(query) { permitSearchKey(query).split(Regex("\\s+")).filter { it.isNotEmpty() } }
    val visible = catalog.filter { item ->
        (sector == null || sector in item.sectors || "Tüm sektörler" in item.sectors) &&
            (job == null || job in item.jobs) && terms.all { it in permitSearchKey(item.searchText) }
    }

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState())
        .padding(horizontal = 16.dp).padding(top = 12.dp, bottom = 24.dp + novaTabBarInset),
        verticalArrangement = Arrangement.spacedBy(12.dp)) {
        NovaListHeading("Çalışma İzni Örnekleri", onBack)
        NovaListHint("56 düzenlenebilir Word örneği. Uygun formu arayın, indirin ve kendi saha prosedürünüze göre uyarlayın. Bu örnekler çalışma onayı veya izin kaydı oluşturmaz.")
        NovaSearchCapsule(query, "Form, iş veya kelime ara", "permit.search") { query = it }
        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            NovaChooserButton("Sektör", sector ?: "Tüm sektörler", "permit.sector", Modifier.weight(1f), open = openFilter == "sector") {
                openFilter = if (openFilter == "sector") null else "sector"
            }
            NovaChooserButton("İş grubu", job ?: "Tüm işler", "permit.job", Modifier.weight(1f), open = openFilter == "job") {
                openFilter = if (openFilter == "job") null else "job"
            }
        }
        if (openFilter == "sector") NovaChooserPanel(listOf(NovaChooserOption(null, "Tüm sektörler")) + sectors.map { NovaChooserOption(it, it) },
            sector, "permit.sector.options") { sector = it; openFilter = null }
        if (openFilter == "job") NovaChooserPanel(listOf(NovaChooserOption(null, "Tüm işler")) + jobs.map { NovaChooserOption(it, it) },
            job, "permit.job.options") { job = it; openFilter = null }
        NovaListSectionHeading("Örnek Formlar", "${visible.size} / ${catalog.size} örnek form")
        message?.let { NovaText(it, style = NovaTypeToken.meta) }
        when {
            catalog.isEmpty() -> NovaEmptyState("Formlar yüklenemedi", "Uygulama paketindeki Word kataloğu bulunamadı.")
            visible.isEmpty() -> NovaEmptyState("Uygun form bulunamadı", "Arama kelimesini veya filtreleri değiştirin.")
            else -> visible.forEach { item ->
                NovaCard(Modifier.fillMaxWidth(), padding = 12) {
                    Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(10.dp), verticalAlignment = Alignment.CenterVertically) {
                        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(4.dp)) {
                            NovaText("${item.code} · ${item.title}", style = NovaTypeToken.bodyStrong, maxLines = 2)
                            NovaText(item.usage, style = NovaTypeToken.meta, maxLines = 1)
                            NovaText(item.jobs.joinToString(" · "), style = NovaTypeToken.metaQuiet, maxLines = 1)
                        }
                        IconButton(onClick = { message = null; pending = item; save.launch(item.filename) },
                            modifier = Modifier.size(44.dp).clip(RoundedCornerShape(12.dp))
                                .background(NovaColorToken.surfaceMuted.color())
                                .semantics { contentDescription = "${item.title} Word dosyasını indir" }
                                .testTag("permit.download.${item.code}")) {
                            NovaIcon("arrow.down.doc", 20.dp)
                        }
                    }
                }
            }
        }
    }
}
